spec/arrays.dd

```diff
@@ -1031,13 +1022,29 @@ immutable(char)[] str5 = str1.idup; // ok, make immutable copy
 
 $(SPEC_RUNNABLE_EXAMPLE_FAIL
 ---------
-char[] str1 = "abc";     // error, "abc" is not mutable
-char[] str2 = "abc".dup; // ok, make mutable copy
+char[] str1 = "abc";     // error, cannot implicitly convert expression `"abc"` of type `string` to `char[]`
+char[] str2 = "abc".dup; // ok, makes mutable copy
 string str3 = "abc";     // ok
-string str4 = str1;      // error, str4 is not mutable
-string str5 = str1.idup; // ok, make immutable copy
+string str4 = str2;      // error, cannot implicitly convert expression `str2` of type `char[]` to `string`
+string str5 = str3;      // ok, makes a mutable str5 with immutable aray contents
+string str6 = str2.idup; // ok, makes immutable copy
 ---------
 )
+        $(P
+        The type immutable(char)[] represents an array of immutable chars. However, the reference to the string is 
```

Comment:
please highlight the compiler keyword : immutable(char)[], immutable string etc.